upload.go

```diff
@@ -0,0 +1,87 @@
+package jwplatform
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+// postFile creates a form file and posts it
+func postFile(filename string, targetUrl string) (*http.Response, error) {
+	bodyBuf := &bytes.Buffer{}
+	bodyWriter := multipart.NewWriter(bodyBuf)
+
+	fileWriter, err := bodyWriter.CreateFormFile("file", filename)
+	if err != nil {
+		return nil, err
+	}
+
+	fh, err := os.Open(filename)
```

Comment:
Might consider using https://github.com/mitchellh/go-homedir or similar to handle expanding `~`, or clearly document that `filename` needs to be an absolute path.